### PR TITLE
xjc generation changes

### PR DESCRIPTION
**Pull request by ds58**
- We now commit the generated xjc files to vcs
- We now call cppbuild.bash on gradle import
- We now generate the xjc files via command line `xjc`, I found this to be less flaky overall. It will not run if xjc is not available on the system.

### CODE REVIEW

File: cppbuild.bash [hunk 1]
```diff
@@ -16,6 +16,34 @@ cd $REPO_ROOT
...
+    # Replace javax.xml.* with jakarta.xml.*, but ignore javax.xml.namespace.QName
+    # Replace @javax.xml.bind.annotation.* with @jakarta.xml.bind.annotation.*
+    sed -i '
+        /import javax\.xml\.namespace\.QName/!s/import javax\.xml\./import jakarta.xml./g
+        s/@javax\.xml\.bind\.annotation\./@jakarta.xml.bind.annotation./g
+        s/javax\.xml\.bind\.annotation\.XmlNsForm/jakarta.xml.bind.annotation.XmlNsForm/g
+    ' "$file"
+
+    # Delete the Generated on: line so we don't constantly have new vcs changes to these files
+    sed -i '/\/\/ Generated on:/d' "$file"
```
Comment (calvertdw) [inline cppbuild.bash]: :pray:

File: cppbuild.bash [hunk 2]
```diff
@@ -16,6 +16,34 @@ cd $REPO_ROOT
 #### Apply patches ####
 patch $REPO_ROOT/ihmc-pub-sub/thirdparty/Fast-RTPS/resources/xsd/fastRTPS_profiles.xsd $REPO_ROOT/ihmc-pub-sub/patches/fastRTPS_profiles.patch
 
+# Generate Java from eprosima XML
```
Comment (calvertdw) [inline cppbuild.bash:19]: :tada:

File: ihmc-pub-sub/build.gradle.kts [hunk 3]
```diff
@@ -8,67 +11,65 @@ ihmc {
 
    configureDependencyResolution()
    javaDirectory("main", "../../swig/FastRTPS/generated")
-   javaDirectory("main", "../../build/generated/sources/xjc/java/main")
+   javaDirectory("xjc", "../xjc/java")
```
Comment (calvertdw) [inline ihmc-pub-sub/build.gradle.kts]: I don't think you need this since you're using a source set.

File: ihmc-pub-sub/build.gradle.kts [hunk 4]
```diff
@@ -8,67 +11,65 @@ ihmc {
...
+   api("com.sun.xml.bind:jaxb-impl:4.0.5") // Match this version with YoVariables
+}
 
-// Cookie cutter function for defining multiple XJC tasks
-fun addXjcTask(taskName: String, schema: String, pkg: String, dest: String) : Task {
-   // If you haven't already, create the generated output dir before running XJC or it will fail
-   file(dest).mkdirs()
+fun runScript(scriptPath: String, envVars: Map<String, String> = emptyMap(), vararg args: String) {
+   val isWindows = System.getProperty("os.name").lowercase().contains("win")
+   val bashCommand = if (isWindows) "C:\\Program Files\\Git\\git-bash.exe" else "bash"
```
Comment (calvertdw) [inline ihmc-pub-sub/build.gradle.kts:37]: Wouldn't it be more reliable to write a powershell?

File: ihmc-pub-sub/build.gradle.kts (hunk 3 above)
Comment (ds58) [reply ihmc-pub-sub/build.gradle.kts]: removed

**Review by calvertdw — APPROVED** [review]

File: ihmc-pub-sub/build.gradle.kts (hunk 4 above)
Comment (ds58) [reply ihmc-pub-sub/build.gradle.kts:37]: No, because bash is not available on the default Path

**Review by TomaszTB — APPROVED** [review]

**Pull request merged**